src/components/ClassInfoModal.tsx

```diff
@@ -133,41 +136,82 @@ export const ClassInfoModal: React.FC<ClassInfoModalProps> = ({
                 </ModalBody>
 
                 <ModalFooter>
-                    <Link
-                        href={
-                            me
-                                ? me.role === roles.VOLUNTEER
-                                    ? `/volunteer/enrollment?classId=${classInfo.id}`
-                                    : `/parent/enrollment/?classId=${classInfo.id}`
-                                : "/login"
-                        }
-                    >
-                        <Button
-                            bg={colourTheme.colors.Blue}
-                            color={"white"}
-                            mx={"auto"}
-                            my={2}
-                            fontWeight={"200"}
-                            _hover={{
-                                textDecoration: "none",
-                                bg: colourTheme.colors.LightBlue,
-                            }}
-                            _active={{
-                                bg: "lightgrey",
-                                outlineColor: "grey",
-                                border: "grey",
-                                boxShadow: "lightgrey",
-                            }}
-                            _focus={{
-                                outlineColor: "grey",
-                                border: "grey",
-                                boxShadow: "lightgrey",
-                            }}
-                            minW={"100%"}
+                    {isFull ? (
+                        <ModalBody>
+                            <Button
+                                bg={colourTheme.colors.Blue}
+                                color={"white"}
+                                mx={"auto"}
+                                my={2}
+                                fontWeight={"200"}
+                                _hover={{
+                                    textDecoration: "none",
+                                    bg: colourTheme.colors.LightBlue,
+                                }}
+                                _active={{
+                                    bg: "lightgrey",
+                                    outlineColor: "grey",
+                                    border: "grey",
+                                    boxShadow: "lightgrey",
+                                }}
+                                _focus={{
+                                    outlineColor: "grey",
+                                    border: "grey",
+                                    boxShadow: "lightgrey",
+                                }}
+                                minW={"100%"}
+                                onClick={() =>
+                                    createWaitlistRegistration(
+                                        me.parent,
+                                        classInfo.id,
+                                    )
```

Comment:
we need an indicator that they've successfully been added to waitlist -- i was thinking:
1. a pop-up that confirms it?
2. the "add to waitlist" button should be grayed out or something afterwards so they don't try to add to waitlist again (not that it would affect anything functionally but just for flow's sake) 

am wondering if design wants to have a say in this / draw something up if you have the capacity ?? could also be a quick follow-up ticket that i can take on @bekahma @vedantdsn